docs/src/build.md

```diff
@@ -0,0 +1,69 @@
+# Building
+
+To build the complete project including all included crates requires some
+configuration beforehand.
+
+## Install Dependencies
+
+First install the Rust compiler to actually compile the project.  Use your
+package manager or [set up in your local user home](https://rustup.rs/).
+
+```sh
+# # For rpm based distributions (Fedora, CentOS, Rocky, ...)
+# dnf install cargo
+#
+# # For apt based distributions (Debian, Ubuntu, ...)
+# apt update
+# apt install cargo
+#
+# # For Arch Linux and its derivatives
+# pacman -Sy cargo
+```
+
+You'll need atleast version 1.61. Most package manager should deliver this or
+more recent version - rustup will always deliver the most up-to-date version.
+
+Then install the required system dependencies for _bindgen_ and the _JULEA_
+bindings.
+
+### Fedora/RHEL
+
+```sh
+# dnf install glib2 glib2-devel libbson libbson-devel clang make pkgconf
+```
+
+### Ubuntu/Debian/...
+
+```sh
+# apt update
+# apt install libglib2.0-0 libglib2.0-dev libbson-1.0-0 libbson-dev clang make pkg-config
```

Comment:
```suggestion
$ apt update
$ apt install libglib2.0-0 libglib2.0-dev libbson-1.0-0 libbson-dev clang make pkg-config
```

Reply:
I know that the highlighting looks odd when rendering with the shell syntax in markdown as it thinks this starts a comment. But considering that we need root permissions to install with apt, wouldn't changing this from `#` to `$` make the snippet incorrect?

Reply:
Good point, I didn't really consider this. `$ sudo apt ...`? :see_no_evil:

Reply:
Probably the most helpful as people can just copy this^^ Will change it to this.